napari/layers/points/points.py

```diff
@@ -1692,8 +1646,46 @@ def get_ray_intersections(
 
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
-        # get the indices of points in view
-        indices, scale = self._slice_data(self._slice_indices)
+
+        # The new slicing code makes a request from the existing state and
+        # executes the request on the calling thread directly.
+        # For async slicing, the calling thread will not be the main thread.
+        request = self._make_slice_request_internal(
+            self._slice_input, self._slice_indices
+        )
+        response = request()
+        self._set_slice_response(response)
+
+    def _make_slice_request(self, dims) -> _PointSliceRequest:
+        """Make a Points slice request based on the given dims and these data."""
+        slice_input = self._make_slice_input(
+            dims.point, dims.ndisplay, dims.order
+        )
+        # TODO: [see Image]
+        #   For the existing sync slicing, slice_indices is passed through
+        # to avoid some performance issues related to the evaluation of the
+        # data-to-world transform and its inverse. Async slicing currently
+        # absorbs these performance issues here, but we can likely improve
+        # things either by caching the world-to-data transform on the layer
+        # or by lazily evaluating it in the slice task itself.
+        slice_indices = slice_input.data_indices(self._data_to_world.inverse)
```

Comment:
Optional: doesn't have to be in this PR, but I think there are a couple of ways to improve this performance issue (likely for the image path too).

1. Delay execution of world-to-data by passing the data-to-world transform chain to the request.
2. Early return on `data_indices` when the value is trivial (i.e. when `ndisplay` >= `ndim`).

Reply:
I'd be tempted to work these changes in either during or after the integration PR.